test/integration/methods/delete.js

```diff
@@ -27,9 +27,12 @@ run((assert, comment) => {
     store.on(changeEvent, data => {
       assert(find(data[deleteMethod].user, id => id === 3),
         'change event shows deleted ID')
+      assert(find(data[updateMethod].user, (update) => {
+        return update.id === 2 && update.pull.enemies[0] === 3
+      }), 'user 3 is removed from user 2 enemies')
       assert(deepEqual(data[updateMethod].user
         .map(x => x.id).sort((a, b) => a - b),
-        [ 1, 2 ]), 'change event shows updated IDs')
+        [ 1, 2, 2 ]), 'change event shows updated IDs')
```

Comment:
I don't think this is right, it shouldn't show the same ID twice.

Reply:
I’m not at my computer right now but if I remember right, the reason for there to be 2 same id’s is that there are two updates happening to that record (the latter being pulling out the enemies id). I can check this again but I remember the update being correct.

Reply:
Okay so the update that is passed is this (as JSON):

```json
[
  {
    "id": 2,
    "pull": {
      "enemies": [
        3
      ]
    }
  },
  {
    "id": 1,
    "pull": {
      "friends": [
        3
      ]
    }
  },
  {
    "id": 2,
    "pull": {
      "friends": [
        3
      ]
    }
  }
]
```

If the test's purpose is to ensure only the id's that are affected, maybe just compare the unique id's?

```js
assert(deepEqual([ ...new Set(data[updateMethod].user
        .map(x => x.id).sort((a, b) => a - b))],
        [ 1, 2 ]), 'change event shows updated IDs')
```

What do you think?